client/src/js/components/courseEvals/CourseEvals.js

```diff
@@ -36,169 +54,107 @@ const StyledSubHeading = styled("h2")`
-      "term":"fallSem",
-      "title":"Software Engineering B",
-      "title_jp":"ソフトウェア工学B",
-      "instructor":"HAGIMOTO, Junzo/HIRANABE, Kenji/FUKAZAWA, Yoshiaki/WASHIZAKI, Hironori",
-      "instructor_jp":"萩本 順三/平鍋 健児/深澤 良彰/鷲崎 弘宜",
-      "occurrences":[
-        {
-           "day":3,
-           "start_period":5,
-           "end_period":5,
-           "building":"-1",
-           "classroom":"undecided",
-           "location":"-1-undecided"
-        }
-      ],
-      "lang":"JP",
-      "keys":[
-        {
-           "school":"FSE",
-           "key":"2603033019012020260303301926"
-        }
-      ]
-    },
-    {
-      "_id":"26GC02300301202026MC03300926",
-      "course_code": 'INFP31ZL',
-      "year":2020,
-      "term":"fallSem",
-      "title":"Software Engineering",
-      "title_jp":"Software Engineering",
-      "instructor":"ARCAINI, Paolo/KOBAYASHI, Tsutomu/FUKAZAWA, Yoshiaki/HONIDEN, Shinichi/WASHIZAKI, Hironori",
-      "instructor_jp":"アルカイニ パオロ/小林 努/深澤 良彰/本位田 真一/鷲崎 弘宜",
-      "occurrences":[
-        {
-           "day":1,
-           "start_period":3,
-           "end_period":3,
-           "building":"-1",
-           "classroom":"undecided",
-           "location":"-1-undecided"
-        }
-      ],
-      "keywords":[
-        "IPSE"
-      ],
-      "lang":"EN",
-      "keys":[
-        {
-           "school":"FSE",
-           "key":"26GC02300301202026MC03300926"
-        }
-      ]
-    }
-  ];
+const getRelatedCourses = (loadedCourses, courseCode, thisCourseKey, thisCourseTitle) => {
+  const relatedCourseIDs = Object.keys(loadedCourses).filter(courseID => loadedCourses[courseID].code === courseCode && getCourseKey(loadedCourses[courseID]) !== thisCourseKey);
+  const relatedCourses = relatedCourseIDs.map(courseID => loadedCourses[courseID]);
+  const sortedRelatedCourses = relatedCourses.sort((a, b) => levenshtein(thisCourseTitle, a.title) - levenshtein(thisCourseTitle, b.title)).slice(0,3);
+  return sortedRelatedCourses;
 }
 
-const filterCourseEvalsByKey = (evaluations, courseKey) => evaluations.filter(e => e.course_key === courseKey);
+async function getCourseEvalsByKey (courseKey) {
+  try {
+    // get evaluations by courseKeys
+    const res = await axios.get(wasetimeApiStatic.courseEvalsBaseURL + courseKey);
+    return res.data;
+  } catch (error) {
+    console.error(error);
+  }
+}
 
 class CourseEvals extends React.Component {
   state = {
     thisCourse: {},
     relatedCourses: [],
-    courseEvals: [],
     thisCourseEvals: [],
+    relatedCourseEvals: [],
     avgSatisfaction: 0,
     avgDifficulty: 0,
     avgBenefit: 0,
-    isLoaded: false
+    isLoaded: false,
+    isModalOpen: false
   };
 
   async componentDidMount () {
     const courseID = qs.parse(this.props.location.search, { ignoreQueryPrefix: true }).courseID;
-    const searchLang = qs.parse(this.props.location.search, { ignoreQueryPrefix: true }).searchLang;
+    let loadedCourses = loadState().fetchedCourses.byId;
```

Comment:
We should remove `loadState()`

Reply:
fetchedCourses.byId in redux store is an object, but fetchedCourses passed by withFetchCourses is an array, so I cannot get the course just with `loadedCourses[courseID]` if I use fetchedCourses passed by withFetchCourses instead of that from `loadState()`.